packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -1060,3 +1125,140 @@ contract OptimismPortalResourceFuzz_Test is CommonTest {
+
+        uint64 gasLimit = optimismPortal.minimumGasLimit(120_001);
+
+        vm.expectRevert("OptimismPortal: data too large");
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), 0, 0, gasLimit, false, data);
+    }
+
+    /// @dev Tests that `depositERC20Transaction` reverts when not enough of the token is approved.
+    function test_depositERC20Transaction_notEnoughAmount_reverts() external {
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+        vm.expectRevert();
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), 1, 0, 0, false, "");
+    }
+
+    /// @dev Tests that `balance()` returns the correct balance when the gas paying token is not ether.
+    function testFuzz_balance_nonEther_succeeds(uint256 _amount) external {
+        // Mint the token to the contract and approve the token for the portal
+        token.mint(address(this), _amount);
+        token.approve(address(optimismPortal), _amount);
+
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), _amount, 0, optimismPortal.minimumGasLimit(0), false, "");
+
+        // Check that the balance has been correctly updated
+        assertEq(optimismPortal.balance(), _amount);
+    }
+
+    /// @dev Tests that `finalizeWithdrawalTransaction` succeeds.
+    function test_finalizeWithdrawalTransaction_provenWithdrawalHash_nonEther_succeeds() external {
+        // Mint the token to the contract and approve the token for the portal
+        token.mint(address(this), _defaultTx.value);
+        token.approve(address(optimismPortal), _defaultTx.value);
+
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(
+            address(bob), _defaultTx.value, 0, optimismPortal.minimumGasLimit(0), false, ""
+        );
+
+        assertEq(optimismPortal.balance(), _defaultTx.value);
+
+        vm.expectEmit(true, true, true, true);
+        emit WithdrawalProven(_withdrawalHash, alice, bob);
+        optimismPortal.proveWithdrawalTransaction(_defaultTx, _proposedOutputIndex, _outputRootProof, _withdrawalProof);
+
+        vm.warp(block.timestamp + l2OutputOracle.FINALIZATION_PERIOD_SECONDS() + 1);
+        vm.expectEmit(true, true, false, true);
+        emit WithdrawalFinalized(_withdrawalHash, true);
+
```

Comment:
We should also be sure to `expectCall` on erc20 token transfer since that is how the balance transfer happens